### ISSUE TITLE
[windows][docker][core] Inconsistency between "ray status" and python ray.available_resources() results on manual cluster

### ISSUE THREAD

**Issue by dkorolov** [ray-project/ray#16909]
### Problem description
I use a manual on-premise cluster and I see different cluster resources availability results with `ray status` command and python `ray.available_resources()` function

1) If I use the `ray status` command on a simple 2 node cluster - I see the only head machine:
> Node status
>  ------------------------------------------------------------'
>  1 node(s) with resources: {'CPU': 1.0, 'node:192.168.140.15': 1.0, 'object_store_memory': 2459790950.0, 'memory': 4919581902.0}

2) If I use `ray.available_resources()` on the same cluster - I see both machines in available resources:
> {'node:192.168.140.15': 1.0, 'memory': 26995238504.0, 'node:192.168.66.3': 1.0, 'object_store_memory': 11920786636.0, 'CPU': 25.0}

I am not sure why such inconsistency can happen.  I assume that I should get the same result from ray status  and python  ray.available_resources()

I use a non-standard cluster.  Head node is a Linux machine (Ubuntu 20.04.2 LTS), I start all manually for testing. I use ray-ml Docker [rayproject/ray-ml:latest-gpu](https://hub.docker.com/r/rayproject/ray-ml) for both head and worker machines, with Ray 1.4.0 and Python 3.7.7 . But for the Worker machine, I use Windows 10 WSL2 Ubuntu linux started as same rayproject/ray-ml:latest-gpu Docker container(I will provide details below). We try to use Windows WSL2 because of company business specifics. 
 
I test this issue with standard tune_basic_example [https://docs.ray.io/en/master/tune/examples/tune_basic_example.html](https://docs.ray.io/en/master/tune/examples/tune_basic_example.html) .  
I try to use both start option `ray.init(address='auto')` and `ray.util.connect()`  but have same inconsistency.  
For me, this cluster never works normally and able to utilize only the head node, and never a second machine. Even when the second machine visible on the dashboard and has no errors when starting.

### Reproduction:
1) I start ray-ml Docker container and Ray head node on Linux machine
```
# start Docker
sudo docker run --shm-size=16GB -t -i --network host  rayproject/ray-ml:latest-gpu
# start ray head
ulimit -n 65536; ray start --head --dashboard-host 0.0.0.0
```
2) I start Worker node. I have Windows machine with Docker Desktop and WSL2 installed. I start ray-ml Docker container from Windows PowerShell 
```
#on Local Machine Windows machine
docker run --shm-size=16GB -t -i --network host  rayproject/ray-ml:latest-gpu
# start worker machine
ulimit -n 65536; ray start --address=192.168.140.15:6379 --redis-password='5241590000000000'
```
I check Ray dashboard and see both head and worker nodes.
After that, I run python (from Worker machine, from the same container command line):
```
import ray
ray.init(address='192.168.140.15:6379', _redis_password='5241590000000000')
print("Ray available_resources:")
print(ray.available_resources())
```
I see both machines in available resources:
> {'node:192.168.140.15': 1.0, 'memory': 26995238504.0, 'node:192.168.66.3': 1.0, 'object_store_memory': 11920786636.0, 'CPU': 25.0}

But if I run in command line on Worker machine (from the same container command line):
```
ray status
```
I see only head node:
> Node status
>  ------------------------------------------------------------'
>  1 node(s) with resources: {'CPU': 1.0, 'node:192.168.140.15': 1.0, 'object_store_memory': 2459790950.0, 'memory': 4919581902.0}

**Comment by dkorolov** [comment]
Also when I run standard tune_basic_example https://docs.ray.io/en/master/tune/examples/tune_basic_example.html . My work just pending and do nothing.  What strange - requested resources is zero. This related to cluster - all work fine on one local node.
I got log like this:


> 2021-07-06 15:04:18,010 WARNING function_runner.py:545 -- Function checkpointing is disabled. This may result in unexpected behavior when using checkpointing features or certain schedulers. To enable, set the train function arguments to be `func(config, checkpoint_dir=None)`.
> == Status ==
> Memory usage on this node: 2.1/7.7 GiB
> Using FIFO scheduling algorithm.
> Resources requested: 0/25 CPUs, 0/0 GPUs, 0.0/26.46 GiB heap, 0.0/11.67 GiB objects
> Result logdir: /home/ray/ray_results/easy_objective_2021-07-06_15-04-18
> Number of trials: 27/100 (27 PENDING)
> +----------------------------+----------+-------+--------------+-----------+-----------+
> | Trial name                 | status   | loc   | activation   |    height |     width |
> |----------------------------+----------+-------+--------------+-----------+-----------|
> | easy_objective_1bfcc_00000 | PENDING  |       | relu         |  36.7811  |  8.81744  |
> | easy_objective_1bfcc_00001 | PENDING  |       | tanh         | -58.2327  |  8.28668  |
> | easy_objective_1bfcc_00002 | PENDING  |       | relu         |  33.0177  |  8.34864  |
> | easy_objective_1bfcc_00003 | PENDING  |       | tanh         | -34.7617  |  9.13621  |
> | easy_objective_1bfcc_00004 | PENDING  |       | relu         | -11.6281  |  8.25981  |
> | easy_objective_1bfcc_00005 | PENDING  |       | tanh         | -92.8813  |  0.178673 |
> | easy_objective_1bfcc_00006 | PENDING  |       | relu         | -95.2377  |  9.20627  |
> | easy_objective_1bfcc_00007 | PENDING  |       | tanh         |  60.3704  |  0.660496 |
> | easy_objective_1bfcc_00008 | PENDING  |       | relu         | -19.9299  |  2.50661  |
> | easy_objective_1bfcc_00009 | PENDING  |       | tanh         |  35.5229  | 11.6106   |
> | easy_objective_1bfcc_00010 | PENDING  |       | relu         |  20.7539  | 17.6791   |
> | easy_objective_1bfcc_00011 | PENDING  |       | tanh         |  67.9049  | 12.5725   |
> | easy_objective_1bfcc_00012 | PENDING  |       | relu         |   4.76204 | 11.6114   |
> | easy_objective_1bfcc_00013 | PENDING  |       | tanh         |  71.973   | 15.819    |
> | easy_objective_1bfcc_00014 | PENDING  |       | relu         | -61.5827  |  0.532706 |
> | easy_objective_1bfcc_00015 | PENDING  |       | tanh         | -29.5506  | 15.3306   |
> | easy_objective_1bfcc_00016 | PENDING  |       | relu         |  43.8618  | 10.2472   |
> | easy_objective_1bfcc_00017 | PENDING  |       | tanh         |  80.6357  | 15.6129   |
> | easy_objective_1bfcc_00018 | PENDING  |       | relu         | -64.6145  | 18.202    |
> | easy_objective_1bfcc_00019 | PENDING  |       | tanh         |  20.8735  | 16.509    |
> +----------------------------+----------+-------+--------------+-----------+-----------+
> ... 7 more trials not shown (7 PENDING)

where `== Status ==` and following part repeat forever

**Comment by DmitriGekhtman** [comment]
Maybe something is going wrong at the windows layer.
cc @ijrsvt for any docker-related insights
cc @wuisawesome for any ray-core/networking insights

Adding a windows tag to description -- there's much room for improvement for ray's windows support.

**Comment by ijrsvt** [comment]
@wuisawesome do you know the difference in data sources for these two codepaths?

**Comment by wuisawesome** [comment]
They both ultimately get their data from GcsResourceManager. 

ray.available_resources() uses `global_state_accessor.get_all_available_resources()`

and `ray status` comes from internal kv, which is set by the autoscaler when it polls `GetAllResourceUsage`.

Actually, this makes me wonder, @dkorolov when you look at `ray status` it should tell you the time of the last update. Does it look up to date? (within 10s)

**Comment by dkorolov** [comment]
@wuisawesome  I checked if `ray status` really updated - all fine I see time changes. It looks up to date. 
I made some extra tests and we set up on a machine with Windows dual boot with Ubuntu Linux.  In native Linux, all works fine.
So the issue is related to the case of running under Windows WSL2 linux.  Technically Ray runs under Linux under this case.  But this looks like a specific case and Ray not always ready for this. It looks Windows WSL2 linux works pretty normal with Ray in local mode,  but I have no success connecting and work in a cluster any way I tried.
 Also I think _autodetect_num_gpus() works not correctly for Windows WSL2 Linux https://github.com/ray-project/ray/blob/master/python/ray/resource_spec.py#L221 . It does not detect GPU when PyTorch or TensorFlow does.
